pincer/cog.py

```diff
@@ -0,0 +1,171 @@
+# Copyright Pincer 2021-Present
+# Full MIT License can be found in `LICENSE` at the project root.
+
+from __future__ import annotations
+from asyncio import ensure_future
+
+from importlib import reload, import_module
+from types import ModuleType
+from typing import TYPE_CHECKING, List
+
+from . import client as _client
+from .commands.chat_command_handler import ChatCommandHandler
+from .commands.interactable import Interactable
+from .exceptions import CogAlreadyExists, CogNotFound
+
+if TYPE_CHECKING:
+    from typing import Type
+    from .client import Client
+
+
+def get_cog_name(cog: Type[Cog]) -> str:
+    """Gets the path to import a cog. This is used as a unique identifier"""
+    try:
+        return f"{cog.__module__}.{cog.__name__}"
+    except AttributeError:
+        return f"{cog.__module__}.{cog.__class__.__name__}"
```

Comment:
Property doesn't really work here cause you need to be able to call this from the class. I'm gonna do a `classmethod` since that seems the easiest and i don't think its a good idea to touch metaclasses and stuff for this.